clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -170,6 +190,20 @@ public synchronized void update(Cluster cluster, long now) {
         this.lastSuccessfulRefreshMs = now;
         this.version += 1;
 
+        if (topicExpiryEnabled) {
+            // Handle expiry of topics from the metadata refresh set.
+            for (Iterator<Map.Entry<String, Long>> it = topics.entrySet().iterator(); it.hasNext(); ) {
+                Map.Entry<String, Long> entry = it.next();
+                Long expireMs = entry.getValue();
```

Comment:
`expireMs` should probably be a `long` since we use it as if it's never null a few lines below.